sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -406,19 +414,40 @@ Scheduler::~Scheduler() {
           "not all resources were released. Please be sure that all kernels "
           "have synchronization points.\n\n");
   }
+  // Please be aware that releaseResources should be called before deletion of
+  // Scheduler. Otherwise there can be the case when objects Scheduler keeps as
+  // fields may need Scheduler for their release and they work with Scheduler
+  // via GlobalHandler::getScheduler that will create new Scheduler object.
+  // Still keep it here but it should do almost nothing if releaseResources
+  // called before.
+  releaseResources();
+}
+
+void Scheduler::releaseResources() {
   // There might be some commands scheduled for post enqueue cleanup that
   // haven't been freed because of the graph mutex being locked at the time,
   // clean them up now.
   cleanupCommands({});
+  DefaultHostQueue.reset();
```

Comment:
It seems to be unsafe to release DefaultHostQueue here.

Reply:
lets first get results if the solution helps